webapp/src/components/code_review/code_review.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+
+import {FileInfo} from 'mattermost-redux/types/files';
+import {Client4} from 'mattermost-redux/client';
+import {Theme} from 'mattermost-redux/types/preferences';
+import hlJS from 'highlight.js';
+import {getFileUrl} from "mattermost-redux/utils/file_utils";
+import { Post, PostList } from 'mattermost-redux/types/posts';
+
+import './code_review.css';
+import CodeComments from 'components/code_comments';
+
+export type Props = {
+    fileInfo: FileInfo;
+    post: Post;
+    onModalDismissed: () => void;
+    theme: Theme;
+    comments: { [key:string]: Post[] }
+};
+
+type State = {
+    code: string;
+    comments?: PostList;
+    lang: string;
+    loading: boolean;
+    success: boolean;
+    prevFileId?: string;
+    line?: number;
+};
+
+const Avatar = (window as any).Components.Avatar;
+const imageURLForUser = (window as any).Components.imageURLForUser;
+
+export default class CodeReview extends React.PureComponent<Props, State> {
+
+    static MAX_FILE_SIZE = 500000;
+
+    static SupportedFiles: { [extension: string]: any } = {
+        diff: {name: 'Diff'},
+        patch: {name: 'Diff'},
+    }
+
+    static getDerivedStateFromProps(props: Props, state: State) {
+        if (props.fileInfo.id !== state.prevFileId) {
```

Comment:
This function would be easier to read if we had an if statement that returned early here:

```suggestion
        if (props.fileInfo.id === state.prevFileId) {
            return null;
        }
```